File: system-test/java/src/main/java/com/optivem/eshop/dsl/core/usecase/external/erp/usecases/base/BaseErpUseCase.java
package com.optivem.eshop.dsl.core.usecase.external.erp.usecases.base;

import com.optivem.eshop.dsl.driver.port.external.erp.ErpDriver;
import com.optivem.eshop.dsl.core.shared.BaseUseCase;
import com.optivem.eshop.dsl.core.shared.UseCaseContext;

public abstract class BaseErpUseCase<TResponse, TVerification> extends BaseUseCase<ErpDriver, TResponse, TVerification> {
    protected BaseErpUseCase(ErpDriver driver, UseCaseContext context) {
        super(driver, context);
    }
}

File: system-test/typescript/src/testkit/dsl/core/usecase/external/erp/usecases/base/BaseErpUseCase.ts
import type { ErpDriver } from '../../../../../../../driver/port/external/erp/erp-driver.js';
import { BaseUseCase } from '../../../../../shared/base-use-case.js';

export abstract class BaseErpUseCase<TResponse, TVerification> extends BaseUseCase<
  ErpDriver,
  TResponse,
  TVerification
> {}
